pkg/hooks/manager.go

```diff
@@ -348,7 +348,7 @@ func (m *hookManager) PreCloudInitIso(vmi *v1.VirtualMachineInstance, cloudInitD
 			}
 			return preCloudInitIsoValidateResult(cloudInitData.DataSource, result.GetCloudInitData(), result.GetCloudInitNoCloudSource())
 		default:
-			panic("Should never happen, version compatibility check is done during Info call")
+			log.Log.Errorf("Unsupported callback version: %s", callback.Version)
```

Comment:
The commit can be dropped if we prefer to panic. If we squash it, it would mean we changed a particular behavior that might not be desirable together with other commit that does something else. It would be different if I have a few commits doing "do not panic".

That said, I'd be more inclined to drop this commit instead of squashing it as it is tangent to the goal of the PR.

Reply:
> I'd be more inclined to drop this commit instead of squashing it as it is tangent to the goal of the PR.

Up to you if you want to drop or keep, but I agree with what you said in https://github.com/kubevirt/kubevirt/pull/10046#discussion_r1374833358 - A bad sidecar shouldn't panic virt-launcher